### PR TITLE
Lockdown Microsoft.ML.Onnx public surface

### LINKED ISSUE #2273: Lockdown Microsoft.ML.Onnx public surface
We ought to internalize as much of that assembly to those things we want to be public, and finalize its public API. This may be completed already; the public surface area I hope should just be the extension methods to export.

### PR DESCRIPTION
**Pull request by codemzs**
fixes #2273

**Reduces public API count from 423 to 1.**

| Before | After |   
|:-:|:-:|
| image | image |

### CODE REVIEW

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public void SimpleEndToEndOnnxConversionTest()
             var transformedData = model.Transform(data);
 
             // Step 2: Convert ML.NET model to ONNX format and save it as a file.
-            var onnxModel = mlContext.Model.ConvertToOnnx(model, data);
+            var onnxModel = mlContext.Model.ConvertToOnnxProtobufModel(model, data);
```
Comment (wschin) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs]: >ConvertToOnnxProtobufModel [](start = 44, length = 26)

Seems ConvertToOnnxProtobuf is better. Model doesn't look like a part of a format's name.

**Comment by wschin** [comment]
>     {

Is it auto-generated?

---
Refers to: src/Microsoft.ML.OnnxConverter/OnnxMl.cs:14 in 4cd1d45. [](commit_id = 4cd1d4538ab9578a6c791efd9b697221d483c5d1, deletion_comment = False)

**Comment by codemzs** [comment]
>     {

I added that piece of code. 

---
In reply to: [465343652](https://github.com/dotnet/machinelearning/pull/2625#issuecomment-465343652) [](ancestors = 465343652)

---
Refers to: src/Microsoft.ML.OnnxConverter/OnnxMl.cs:14 in 4cd1d45. [](commit_id = 4cd1d4538ab9578a6c791efd9b697221d483c5d1, deletion_comment = False)

**Comment by wschin** [comment]
>     {

I guess there is no way to have protobuf complier to do this for us, right? If so, please add a note to OnnxML.md so that future maintainers can see how this file got created.

---
In reply to: [465344411](https://github.com/dotnet/machinelearning/pull/2625#issuecomment-465344411) [](ancestors = 465344411,465343652)

---
Refers to: src/Microsoft.ML.OnnxConverter/OnnxMl.cs:14 in 4cd1d45. [](commit_id = 4cd1d4538ab9578a6c791efd9b697221d483c5d1, deletion_comment = False)

**Pull request merged**